src/Microsoft.Azure.WebJobs.Host/Hosting/JobHostService.cs

```diff
@@ -13,11 +13,13 @@ public class JobHostService : IHostedService
     {
         private readonly ILogger<JobHostService> _logger;
         private readonly IJobHost _jobHost;
+        private readonly IHostedService _jobHostHostedService;
 
         public JobHostService(IJobHost jobhost, ILogger<JobHostService> logger)
         {
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _jobHost = jobhost;
+            _jobHostHostedService = jobhost as IHostedService;
```

Comment:
Can you do this cast down in `StopAsync` so we don't need a separate field?

Reply:
Sure.